src/common/snippets/include/snippets/runtime_configurator.hpp

```diff
@@ -109,12 +120,70 @@ class RuntimeConfigurator {
     void update_buffer_scratchpad_size(const lowered::LinearIRPtr& linear_ir) const;
     /**
      * @brief Calculate data offsets of LinearIR and update these values in RuntimeConfig
+     * @param shapes shapes used in offsets computation
+     * @param layouts layouts used in offsets computation
+     */
+    void update_data_offsets(const std::vector<ov::snippets::VectorDims>& shapes,
+                             const std::vector<std::vector<size_t>>& layouts) const;
+    /**
+     * @brief Extract shapes from m_io_descs
      */
-    void update_data_offsets() const;
+    std::vector<ov::snippets::VectorDims> extract_shapes();
```

Comment:
should be these methods const?
Moreover you can unite these method and return tuple or pair:
`std::tie(shapes, layouts) = extract_shapes_and_layouts()`

Reply:
Thanks for the suggestion, now the method are const. As for method combination: I don't think that we will benefit from that: we will spend even more code lines in this case (2 on empty shapes/layouts initialization + `extract_shapes_and_layouts()` call)

Reply:
But we don't need to iterate through all descriptors twice 😄 
However, I don't insist